Wrappers/Python/cil/optimisation/algorithms/FISTA.py

```diff
@@ -93,43 +101,53 @@ class ISTA(Algorithm):
     """
 
     def _provable_convergence_condition(self):
-        return self.step_size <= 0.99*2.0/self.f.L
+        if isinstance(self.step_size_rule, ConstantStepSize): 
+            return self.step_size_rule.step_size <= 0.99*2.0/self.f.L
+        else:
+            raise TypeError("Can't check convergence criterion for non-constant step size")
 
     @property
     def step_size(self):
-       return self._step_size
+        if isinstance(self.step_size_rule, ConstantStepSize):
+            return self.step_size_rule.step_size
+        else:
+            raise TypeError("There is not a constant step size, it is set by a step-size rule")
+
 
     # Set default step size
-    def set_step_size(self, step_size):
+    def get_constant_step_size(self, step_size):
         """ Set default step size.
         """
 
         if step_size is None:
             if isinstance(self.f, ZeroFunction):
```

Comment:
Can we run fista/ista with `f=0`?

Reply:
Yes, was added in https://github.com/TomographicImaging/CIL/pull/1580

Reply:
Maybe in the future we add a PRoximal Point Algorithm